contracts/erc20-bridge-sampler/contracts/src/ERC20BridgeSampler.sol

```diff
@@ -431,6 +434,143 @@ contract ERC20BridgeSampler is
         }
     }
 
+    /// @dev Sample sell quotes from an arbitrary on-chain liquidity provider.
+    ///      Calls the provider contract with the default gas limit (200k).
+    /// @param providerAddress Address of the liquidity provider contract.
+    /// @param takerToken Address of the taker token (what to sell).
+    /// @param makerToken Address of the maker token (what to buy).
+    /// @param takerTokenAmounts Taker token sell amount for each sample.
+    /// @return makerTokenAmounts Maker amounts bought at each taker token
+    ///         amount.
+    function sampleSellsFromLiquidityProvider(
+        address providerAddress,
+        address takerToken,
+        address makerToken,
+        uint256[] memory takerTokenAmounts
+    )
+        public
+        view
+        returns (uint256[] memory makerTokenAmounts)
+    {
+        return sampleSellsFromLiquidityProvider(
+            providerAddress,
+            takerToken,
+            makerToken,
+            takerTokenAmounts,
+            DEFAULT_CALL_GAS
+        );
+    }
+
+    /// @dev Sample sell quotes from an arbitrary on-chain liquidity provider.
+    /// @param providerAddress Address of the liquidity provider contract.
+    /// @param takerToken Address of the taker token (what to sell).
+    /// @param makerToken Address of the maker token (what to buy).
+    /// @param takerTokenAmounts Taker token sell amount for each sample.
+    /// @param callGasLimit Gas limit to use for calls to the provider's
+    ///        `getSellQuote` method.
+    /// @return makerTokenAmounts Maker amounts bought at each taker token
+    ///         amount.
+    function sampleSellsFromLiquidityProvider(
```

Comment:
Any reason why we have a gas and non-gas stipened version? Are there plans to call this directly? Would prefer not to have functions we don't use.

Reply:
Yeah we can get rid of one of them; I'd lean towards keeping the one _with_ the `callGasLimit` parameter for similar reasons to the above –– we don't know a priori how much gas to stipend a liquidity provider that might get added